### DISCUSSION TITLE
Upstreaming some internal development?

### DISCUSSION THREAD

**Discussion by ghost** [awslabs/aws-js-s3-explorer#111] [General]
Hi, 

I've been working on some extensions to the alpha-v2 branch of the project as part of an internal project at my company. I'm not sure if there's currently any active development on that branch here, but we were wondering if there might be any interest in us upstreaming our work. The most notable things we've done are add support in the settings menu for alternative S3 endpoints and add configuration options that can be set by whoever deploys the explorer (allowing them to specify a default bucket, etc. if they're anticipating a less technical set of users).

If any of the maintainers of the project would like us to start the process to contribute that work, let me know, and we can get started on it.

Thanks, and thanks for a very useful tool!

**Comment by john-aws** [comment]
Hi, thanks for kicking off this discussion. We're open to considering the inclusion of new developments that are broadly useful to S3 users and that do not conflict with the basic design goal of a simple-to-deploy static site. Happy to hear more.

**Reply by ghost** [reply]
No problem, hopefully we can get things upstreamed if you're interested. To elaborate a little more on what we've done:

- Added an endpoint configuration option to the settings menu
![image](https://user-images.githubusercontent.com/108036834/175394527-b0a9361d-0dea-4893-88b9-91313a0893a9.png)
- Working on adding something like a JSON config file that would go with the current three files and allow configuration of the default endpoint, a default bucket, a default view/prefix, hiding the private bucket options, etc.
    - The plan is roughly to default to what's currently present and allow configuration for use cases like "I have a public bucket that I want non-technicall users to explore; I don't want them to get confused by the options"

Definitely let me know if you're interested. We've got some internal processes we'd have to complete to upstream anything, but if you find it useful we'd be happy to get started on them.

**Reply by john-aws** [reply]
Hi, this does sound like it could work. Ideally the JSON file would be optional to minimize the deployment for vanilla users, and the README would be updated to describe the new, optional feature. Thanks in advance for the potential contribution.

**Reply by ghost** [reply]
Apologies for the very late reply. I ended up getting busy with various projects and have pretty much run out of time to work on upstreaming stuff before the end of my summer internship, so it doesn't look like I'll be able to do anything further on this idea.

I'm sorry for having gotten your hopes up for the features if I did. Although, being pretty new to AngularJS, not sure the code would have been up to standards anyways 😂.
